src/components/codeBlock.tsx

```diff
@@ -106,52 +106,154 @@ type KeywordSelectorProps = {
+  codeKeywords?.PROJECT?.forEach(projectKeyword => {
+    orgSet.add(projectKeyword.ORG_SLUG);
+  });
+  const orgs = [...orgSet];
+
+  const [isAnimating, setIsAnimating] = useState(false);
 
-  // When not signed in, we just show a placeholder, as the user can't generate a token in this case
   if (!codeKeywords.USER) {
+    // User is not logged in - show dummy token
     return <Fragment>sntrys_YOUR_TOKEN_HERE</Fragment>;
   }
 
-  const currentSelectionIdx = sharedSelection.PROJECT ?? 0;
-  const currentSelection = choices[currentSelectionIdx];
+  if (tokenState.status === 'success') {
+    return <Fragment>{tokenState.token}</Fragment>;
+  }
 
-  const name = `Generated by Docs for ${currentSelection.PROJECT_SLUG} on ${new Date()
-    .toISOString()
-    .slice(0, 10)}`;
+  if (tokenState.status === 'error') {
+    return <Fragment>There was an error while generating your token.</Fragment>;
+  }
 
-  const updateToken = async () => {
-    if (tokenState !== 'none') {
-      return;
+  if (tokenState.status === 'loading') {
+    return <Fragment>Generating token...</Fragment>;
+  }
+
+  const selector = isOpen && (
+    <PositionWrapper style={styles.popper} ref={setDropdownEl} {...attributes.popper}>
+      <AnimatedContainer>
+        <Dropdown>
+          <Arrow
+            style={styles.arrow}
+            data-placement={state?.placement}
+            data-popper-arrow
+          />
+          <DropdownHeader>Select an organization:</DropdownHeader>
+          <Selections>
+            {orgs.map(org => {
+              return (
+                <ItemButton
+                  key={org}
+                  isActive={false}
+                  onClick={() => {
+                    createToken(org);
+                    setIsOpen(false);
+                  }}
+                >
+                  {org}
+                </ItemButton>
+              );
+            })}
+          </Selections>
+        </Dropdown>
+      </AnimatedContainer>
+    </PositionWrapper>
+  );
+
+  const portal = getPortal();
+
+  const handlePress = () => {
+    if (orgs.length === 1) {
+      createToken(orgs[0]);
+    } else {
+      setIsOpen(!isOpen);
     }
-    setTokenState('loading');
-    const tokenStr = await createOrgAuthToken({
-      orgSlug: currentSelection.ORG_SLUG,
-      name,
-    });
-    setTokenState(token ? 'success' : 'error');
-    setToken(tokenStr);
   };
 
   return (
-    <KeywordDropdown onClick={updateToken}>
-      {tokenState === 'none'
-        ? 'Click to generate token'
-        : tokenState === 'loading'
-        ? 'Generating...'
-        : token
-        ? token
-        : 'Error generating token'}
-    </KeywordDropdown>
+    <Fragment>
+      <KeywordDropdown
+        ref={setReferenceEl}
+        role="button"
+        title="Click to generate token"
+        tabIndex={0}
+        onClick={() => {
+          handlePress();
+        }}
+        onKeyDown={e => {
+          if (e.key === 'Enter') {
```

Comment:
A native button also fires on `Space`.
```suggestion
          if (['Enter', 'Space'].includes(e.key)) {
```